### PR TITLE
Split aad starter with two modules based on different spring boot versions

### PR DESCRIPTION
**Pull request by moarychan**
According to the two major versions of spring boot, split current module azure-active-directory-spring-boot-starter into two modules：
* 2.2.X, artifact id is azure-active-directory-spring-boot-2-2-starter
* 2.3.X, artifact id is azure-active-directory-spring-boot-2-3-starter

### CODE REVIEW

File: sdk/spring/azure-spring-boot-2-2-starter-active-directory/pom.xml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
+      <artifactId>azure-spring-boot</artifactId>
+      <version>2.4.0-beta.1</version> <!-- {x-version-update;com.microsoft.azure:azure-spring-boot;current} -->
+    </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
+      <artifactId>spring-web</artifactId>
+      <version>5.2.8.RELEASE</version> <!-- {x-version-update;org.springframework:spring-web;external_dependency} -->
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-2-2-starter-active-directory/pom.xml:41]: This should be `springboot22_org.springframework:spring-web`? I think spring-boot-2-2 and 2-3 depend on different version of spring-web.

Comment (chenrujun) [reply sdk/spring/azure-spring-boot-2-2-starter-active-directory/pom.xml:41]: we can use `update_version_for_external_dependencies.py` to get versions in spring-boot-2-2

File: sdk/spring/azure-spring-boot-2-2-starter-active-directory/pom.xml [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.springframework.security</groupId>
+      <artifactId>spring-security-core</artifactId>
+      <version>5.3.4.RELEASE</version> <!-- {x-version-update;org.springframework.security:spring-security-core;external_dependency} -->
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-2-2-starter-active-directory/pom.xml:46]: Same to other dependencies related to spring.

File: sdk/spring/azure-spring-boot-2-3-starter-active-directory/src/main/resources/aad.enable.config [hunk 3]
```diff
@@ -0,0 +1 @@
+dummy
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-2-3-starter-active-directory/src/main/resources/aad.enable.config:1]: Do we need to add `newline` at end of file?

File: sdk/spring/azure-spring-boot/README.md [hunk 4]
```diff
@@ -41,7 +41,8 @@ Alternatively you may want to use the [starters](https://github.com/Azure/azure-
 ## Key concepts
 This project provides auto-configuration for the following Azure services:
 
-- [Azure Active Directory](../azure-spring-boot-starter-active-directory)
+- [Spring Boot 2.2 for Azure Active Directory](../azure-spring-boot-2-2-starter-active-directory)
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/README.md:44]: I think `Azure Active Directory for Spring Boot 2.2` will make more sense. Same to other items.

**Review by chenrujun — CHANGES_REQUESTED** [review]

**Comment by chenrujun** [comment]
Hi, @moarychan , please resolve conflicts. 🙏

**Comment by moarychan** [comment]
Will create new PR for this change.

**Comment by moarychan** [comment]
Hi @chenrujun, I will update the code based on your proposal, but not in the current branch, all starter versions are changed together in a new branch.

**Pull request closed without merging**